driver-core/src/main/com/mongodb/internal/dns/DnsResolver.java

```diff
@@ -16,29 +16,14 @@
 
 package com.mongodb.internal.dns;
 
-import com.mongodb.MongoClientException;
-import com.mongodb.MongoConfigurationException;
-
-import javax.naming.Context;
-import javax.naming.NamingEnumeration;
-import javax.naming.NamingException;
-import javax.naming.directory.Attribute;
-import javax.naming.directory.Attributes;
-import javax.naming.directory.InitialDirContext;
-import java.util.ArrayList;
-import java.util.Hashtable;
 import java.util.List;
 
-import static java.lang.String.format;
-import static java.util.Arrays.asList;
-
 /**
- * Utility class for resolving SRV and TXT records.
+ * Utility interface for resolving SRV and TXT records.
  *
  * <p>This class should not be considered a part of the public API.</p>
  */
-public final class DnsResolver {
-
+public interface DnsResolver {
     // The format of SRV record is
```

Comment:
Method docs formatting.

Reply:
Removed the comments, as this interface is internal and the comments are duplicated in the implementation (which seems a more appropriate place to have the comments since they are about the implementation).